src/main/java/svc/managers/SMSManager.java

```diff
@@ -0,0 +1,248 @@
+package svc.managers;
```

Comment:
There are no tests for this at all. There is a lot of code here and to have no tests is not something we can have. Don't worry about being exhaustive but please handle the primary paths.